source/Interface.cpp

```diff
@@ -618,19 +650,98 @@ void Interface::TextElement::Draw(const Rectangle &rect, const Information &info
 
 
 
-// Add any click handlers needed for this element. This will only be
-// called if the element is visible and active.
-void Interface::TextElement::Place(const Rectangle &bounds, Panel *panel) const
+// Members of the WrappedElement class:
+
+// Constructor.
+Interface::WrappedTextElement::WrappedTextElement(const DataNode &node, const Point &globalAnchor)
 {
-	if(buttonKey && panel)
-		panel->AddZone(bounds, buttonKey);
+	if(node.Size() < 2)
+		return;
+
+	isDynamic = (node.Token(0) == "wrapped string" || node.Token(0) == "wrapped dynamic button");
+	if(node.Token(0) == "wrapped button" || node.Token(0) == "wrapped dynamic button")
+	{
+		buttonKey = node.Token(1).front();
+		if(node.Size() >= 3)
+			str = node.Token(2);
+	}
+	else
+		str = node.Token(1);
+
+	// This function will call ParseLine() for any line it does not recognize.
+	Load(node, globalAnchor);
+
+	// Initialize the WrappedText.
+	text.SetAlignment(alignment);
+	text.SetTruncate(truncate);
+	text.SetWrapWidth(Bounds().Width());
+
+	FinishLoadingColors();
 }
 
 
 
-string Interface::TextElement::GetString(const Information &info) const
+// Parse the given data line: one that is not recognized by Element
+// itself. This returns false if it does not recognize the line, either.
+bool Interface::WrappedTextElement::ParseLine(const DataNode &node)
 {
-	return isDynamic ? info.GetString(str) : str;
+	if(node.Token(0) == "size" && node.Size() >= 2)
+		fontSize = node.Value(1);
+	else if(node.Token(0) == "color" && node.Size() >= 2)
+		color[Element::ACTIVE] = GameData::Colors().Get(node.Token(1));
+	else if(node.Token(0) == "inactive" && node.Size() >= 2)
+		color[Element::INACTIVE] = GameData::Colors().Get(node.Token(1));
+	else if(node.Token(0) == "hover" && node.Size() >= 2)
+		color[Element::HOVER] = GameData::Colors().Get(node.Token(1));
+	else if(node.Token(0) == "truncate" && node.Size() >= 2)
+	{
+		if(node.Token(1) == "none")
+			truncate = Truncate::NONE;
+		else if(node.Token(1) == "front")
+			truncate = Truncate::FRONT;
+		else if(node.Token(1) == "middle")
+			truncate = Truncate::MIDDLE;
+		else if(node.Token(1) == "back")
+			truncate = Truncate::BACK;
+		else
+			return false;
+	}
+	else if(node.Token(0) == "alignment")
+	{
+		if(node.Token(1) == "left")
+			alignment = Alignment::LEFT;
+		else if(node.Token(1) == "center")
+			alignment = Alignment::CENTER;
+		else if(node.Token(1) == "right")
+			alignment = Alignment::RIGHT;
+		else if(node.Token(1) == "justified")
+			alignment = Alignment::JUSTIFIED;
+		else
+			return false;
+	}
+	else
+		return false;
+
+	return true;
+}
```

Comment:
This appears very similar to `BasicTextELement::ParseLine`, and I think most of the fields being configured here are members of `TextElement`. Maybe `BasicTextELement::ParseLine` should be a virtual member of `TextElement` with `WrappedElement::ParseLine` overriding it, only checking for data relating to members of `WrappedElement`, not its parent, and calling the parents version of the method.